File: algorithm/src/main/java/baekjoon/p04447/Main.java
package baekjoon.p04447;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        while (n-- > 0) {
            System.out.println(solve(sc.nextLine()));
        }
    }

    private static String solve(String input) {
        String lower = input.toLowerCase();
        int g = cnt(lower, 'g');
        int b = cnt(lower, 'b');

        if (g > b) {
            return input + " is GOOD";
        } else if (g < b) {
            return input + " is A BADDY";
        } else {
            return input + " is NEUTRAL";
        }
    }

    private static int cnt(String str, char target) {
        int ret = 0;
        for (char ch : str.toCharArray()) {
            if (ch == target) {
                ret++;
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p04447/Main.kt
package baekjoon.p04447

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        println(solve(readLine()!!))
    }
}

private fun solve(input: String): String {
    val lower = input.toLowerCase()
    val g = lower.count { it == 'g' }
    val b = lower.count { it == 'b' }

    return when {
        g > b -> input.plus(" is GOOD")
        g < b -> input.plus(" is A BADDY")
        else -> input.plus(" is NEUTRAL")
    }
}